Sources/StreamChat/ChatClient.swift

```diff
@@ -132,6 +132,12 @@ public class ChatClient {
         environment.callRepositoryBuilder(apiClient)
     }()
 
+    func makeMessagesPaginationStateHandler() -> MessagesPaginationStateHandling {
+        MessagesPaginationThreadSafeDecorator(
+            decoratee: MessagesPaginationStateHandler()
```

Comment:
The parameter name and the object name don't really hint any relation. Can we make it more clear that the object should be passed as this parameter?

Reply:
@ipavlidakis I'm not sure I'm following. Can you clarify?

Reply:
I think he means the name of the parameter: the decorator seems specific to messages pagination, but the parameter name is generic and doesn't provide a clear indication what should be provided.

Reply:
This part of the Decorator pattern, it is meant to be generic, because it can be combined with other decorators. Example:

```swift
MessagesPaginationThreadSafeDecorator(
    decoratee: MessagesPaginationLoggingDecorator(
          decoratee: MessagesPaginationStateHandler() 
    )
}
```

The reason this name is used is for 2 reasons, to easily identify the pattern and to not limit on what the Decorator can decorate, since it can be a concrete implementation, or one more Decorator.

Reply:
I understand your intent, but I find the Decorator Pattern naming in a UI framework, to be confusing. Anyway, if you are going to introduce the Decorator Pattern in the framework, then the `MessagesPaginationThreadSafeDecorator` can be more generic and actually be a reusable component.